proxy/src/control/discovery.rs

```diff
@@ -36,6 +36,12 @@ pub struct Discovery {
 #[derive(Debug)]
 pub struct Watch<B> {
     rx: mpsc::UnboundedReceiver<Update>,
+    /// Map associating addresses with the `Store` for the watch on that
+    /// service's metric labels (as provided by the Destination service).
+    ///
+    /// This is used to update the `Labeled` middleware on those services
+    /// without requiring the service stack to be re-bound.
+    metric_labels: HashMap<SocketAddr, futures_watch::Store<Option<DstLabels>>>,
```

Comment:
obligatory "indexmap?"

Reply:
`hashmap::Entry` was already imported into this scope, and so importing `indexmap::Entry` would require a renaming import. I was going to open a subsquent PR to change everything in the `discovery` module to use `IndexMap`so I didn't have to touch unrelated code in this PR.

Reply:
fine either way